### PR TITLE
DS-8916: Refactoring Templates as per new PDS client

### PR DESCRIPTION
**Pull request by jkagliwal-px**
solves - https://purestorage.atlassian.net/browse/DS-8916

Torpedo Build - https://jenkins.pwx.dev.purestorage.com/job/Users/job/Madan/job/platfrom-build-torpedo/423/

Able to create templates based on DS Under test. 
```
RevisionUid for the ds under test [PostgreSQL] is- [pds:rev:89ebd55d-7827-418e-bb0c-f47a4eddcc47]
2024-04-08 18:05:44 +0530:[INFO] [{TestPlatformTemplates}] templateValue formed is- [map[CpuLimit:2 CpuRequest:1 MemoryLimit:4G MemoryRequest:2G NewStorageSize:5G StorageRequest:1G]]
2024-04-08 18:05:46 +0530:[INFO] [{TestPlatformTemplates}] Create ResourceConfiguration Request formed is- {{{ten:acc:e9272122-9d7e-4c20-b257-bee7ebb9561a 0x1400000fc50} {{ {0 0} {[]} {[]} {[]}  {0 0}}} {} { <nil>} {} {}}}
2024-04-08 18:05:46 +0530:[INFO] [{TestPlatformTemplates}] [api.(*PLATFORM_API_V1).getTemplateClient:#223] - Creating client from PLATFORM_API_V1 package
2024-04-08 18:05:48 +0530:[INFO] [{TestPlatformTemplates}] tenantID fetched is- ten:acc:e9272122-9d7e-4c20-b257-bee7ebb9561a
2024-04-08 18:05:49 +0530:[INFO] [{TestPlatformTemplates}] Template create req formed is- {{{}} 0x14000bf3b68 ten:acc:e9272122-9d7e-4c20-b257-bee7ebb9561a 0x14001342168}
2024-04-08 18:05:51 +0530:[INFO] [{TestPlatformTemplates}] response got is- {0x14000cc3450 0x14000aa4920 0x1400011b068}
2024-04-08 18:05:53 +0530:[INFO] [{TestPlatformTemplates}] App config ID- tmpl:079d7d4b-4cd1-498f-bb5c-6536d069520d
```

### CODE REVIEW

**Review by Madanagopal19 — CHANGES_REQUESTED** [review]

File: drivers/unifiedPlatform/pdsLibs/templateDefinitionLib.go [hunk 1]
```diff
@@ -69,24 +52,12 @@ func CreateServiceConfigTemplate(tenantId string, serviceConfig ServiceConfigura
...
-	templateValue["fg"] = defaultVolGroup
-	templateValue["secure"] = defaultVolSecure
+func CreateStorageConfigTemplate(tenantId string, dsName string, templateConfigs StorageConfiguration) (*automationModels.PlatformTemplatesResponse, error) {
+	revisionUid, err := GetRevisionUidForApplication(dsName)
+	templateName := "pdsAutoSVCTemp" + utilities.RandomString(5)
 
-	templateName := "pdsAutoStTemp" + utilities.RandomString(5)
-	log.InfoD("Temp value formed in lib folder is- [%v]", templateValue)
+	templateValue := structToMap(templateConfigs)
+	log.InfoD("Temp value fromed in lib folder is- [%v]", templateValue)
```
Comment (Madanagopal19) [inline drivers/unifiedPlatform/pdsLibs/templateDefinitionLib.go:60]: nit: formed

**Review by Madanagopal19 — CHANGES_REQUESTED** [review]

**Review by atrivedi-px — APPROVED** [review]
LGTM

**Review by aravindranath14 — APPROVED** [review]

File: drivers/unifiedPlatform/platform/backend/v1/api/templates.go [hunk 2]
```diff
@@ -134,14 +130,13 @@ func (template *PLATFORM_API_V1) DeleteTemplate(templateReq *PlatformTemplatesRe
...
+	log.InfoD("Template create req formed is- %v", templateDelRequest)
+	templateModel, res, err := templateDelRequest.Execute()
+	if err != nil || res.StatusCode != status.StatusOK {
+		return fmt.Errorf("Error when calling `TemplateServiceCreateTemplateExecute`: %v\n.Full HTTP response: %v", err, res)
 	}
-	log.InfoD("Successfully DELETED the template Roles")
-	log.Infof("Value of template - [%v]", templateModel)
-	err = utilities.CopyStruct(templateModel, &templateResponse)
-	log.Infof("Value of template after copy - [%v]", templateResponse)
+	err = utilities.CopyStruct(templateModel, &templateResponse.Create)
```
Comment (Madanagopal19) [inline drivers/unifiedPlatform/platform/backend/v1/api/templates.go:140]: handle the error

**Review by Madanagopal19 — APPROVED** [review]
Minor comment, approving the PR, kindly fix it before merging

**Pull request merged**